### PR TITLE
we can call honest, rule-driven president election 2

### PR DESCRIPTION
**Pull request by akowal3**
# Summary

Implemented compulsory `judge.CallPresidentElection` so that it follows rules in play 🤞 
Left places where our personality/trust could be used to influence the elections.

## Auxilary info
Freaking global rule/variable cache change...

## Test Plan

- Yes.

### CODE REVIEW

File: internal/clients/team4/judge.go [hunk 1]
```diff
@@ -108,29 +108,45 @@ func (j *judge) HistoricalRetributionEnabled() bool {
 // CallPresidentElection is called by the judiciary to decide on power-transfer
 // COMPULSORY: decide when to call an election following relevant rulesInPlay if you wish
 func (j *judge) CallPresidentElection(monitoring shared.MonitorResult, turnsInPower int, allIslands []shared.ClientID) shared.ElectionSettings {
-	// example implementation calls an election if monitoring was performed and the result was negative
-	// or if the number of turnsInPower exceeds 3
+
 	var electionsettings = shared.ElectionSettings{
 		VotingMethod:  shared.Runoff,
```
Comment (tobhil98) [inline internal/clients/team4/judge.go:113]: Why is this hardcoded?

File: internal/clients/team4/utilities.go [hunk 2]
```diff
@@ -19,6 +19,13 @@ func (c *client) getSeason() uint {
 	return 0
 }
 
+func (c *client) getTurnLength(role shared.Role) uint {
+	if c.ServerReadHandle != nil {
+		return c.ServerReadHandle.GetGameConfig().IIGOClientConfig.IIGOTermLengths[role]
+	}
+	return 1
```
Comment (tobhil98) [inline internal/clients/team4/utilities.go]: Maybe return 0, so we know it was not called for whatever reason?

**Review by tobhil98 — APPROVED** [review]
Looks good. 

Minor questions

File: internal/clients/team4/judge.go (hunk 1 above)
Comment (akowal3) [reply internal/clients/team4/judge.go:113]: is that just a question on the comment? it was hardcoded in baseclient implementation, but is not anymore, as it's rule-driven

Comment (akowal3) [reply internal/clients/team4/judge.go:113]: is that just a question on the comment that I removed? it was hardcoded in baseclient implementation, but is not anymore, as it's rule-driven

File: internal/clients/team4/utilities.go (hunk 2 above)
Comment (akowal3) [reply internal/clients/team4/utilities.go]: ok

File: internal/clients/team4/judge.go (hunk 1 above)
Comment (tobhil98) [reply internal/clients/team4/judge.go:113]: Sorry, VotingMethod:  shared.Runoff

Comment (akowal3) [reply internal/clients/team4/judge.go:113]: do we have a better way of deciding whether it should be runoff, instant runoff or the 3rd option?

Comment (akowal3) [reply internal/clients/team4/judge.go:113]: I could do random, but I'm not sure that's better

Comment (tobhil98) [reply internal/clients/team4/judge.go:113]: Do we not need to use the method that's chosen in the rules?

Comment (akowal3) [reply internal/clients/team4/judge.go:113]: there isn't a rule on voting method

**Pull request merged**